apps/webpack.config.js

```diff
@@ -670,18 +659,58 @@ function createWebpackConfig({
               }),
             ]),
       }),
-      ...(watch
+      new PyodidePlugin(),
+      ...(envConstants.HOT
         ? [
-            new LiveReloadPlugin({
-              appendScriptTag: envConstants.AUTO_RELOAD,
-            }),
+            new webpack.HotModuleReplacementPlugin({}),
+            new ReactRefreshWebpackPlugin(),
+            // Prints a URL for accessing the Dashboard via webpack-dev-server
+            {
+              apply: compiler => {
+                compiler.hooks.afterDone.tap('PrintDashboardURL', stats => {
+                  if (stats.hasErrors()) return;
+
+                  if (!process.env.WEBPACK_SERVE) {
+                    console.warn(
+                      "webpack-dev-server should be running, but it doesn't seem to be, url may be wrong"
+                    );
+                  }
+
+                  const TIMEOUT_SO_PRINT_IS_LAST = 1000;
+                  setTimeout(() => {
+                    const WEBPACK_DEV_SERVER_URL = `http://localhost-studio.code.org:${WEBPACK_DEV_SERVER_PORT}`;
+                    const BOLD = '\x1b[1m';
+                    const MAGENTA_BG = `\x1b[45m\x1b[30m${BOLD}`;
+                    const RESET = '\x1b[0m';
+                    console.log(
+                      `\n${MAGENTA_BG}To use webpack-dev-server, access Dashboard at:${RESET} ${BOLD}${WEBPACK_DEV_SERVER_URL}${RESET}`
+                    );
+                  }, TIMEOUT_SO_PRINT_IS_LAST);
+                });
+              },
+            },
           ]
         : []),
-      ...(watch && watchNotify
-        ? [new WebpackNotifierPlugin({alwaysNotify: true})]
-        : []),
-      new PyodidePlugin(),
     ],
+    devServer: envConstants.HOT
+      ? {
+          allowedHosts: ['localhost-studio.code.org'],
+          port: WEBPACK_DEV_SERVER_PORT,
+          proxy: [
+            {
+              context: ['**'],
+              target: 'http://localhost-studio.code.org:3000',
+              changeOrigin: false,
+              logLevel: 'debug',
+            },
+          ],
+          host: '0.0.0.0',
+          hot: true,
```

Comment:
Nice to be explicit here, but I get this warning when running `yarn start`, so maybe remove?:

`<w> [webpack-dev-server] "hot: true" automatically applies HMR plugin, you don't have to add it manually to your webpack configuration.`

Reply:
I believe this is complaining _not_ that we set `hot: true` explicitly here despite it being the default, but instead its complaining that we manually add the HMR plugin in this line:
```
new webpack.HotModuleReplacementPlugin({}),
```

However, not having that line there was triggering a different and slightly scarier warning in react-refresh-webpack-plugin discussed here: https://github.com/code-dot-org/code-dot-org/pull/56882#issuecomment-2002595741

In theory setting hot adds that plugin, but I think its doing so too late for react-refresh-webpack-plugin to detect it. In my reading of the code, I /think/ that react-refresh-webpack-plugin is actually at fault, and this may be a spurious warning, but it was complex enough I didn't feel confident, and given a choice of which warning-that-probably-causes-no-harm to have, I thought this was the less scary. What you think?